source/extensions/filters/common/lua/BUILD

```diff
@@ -21,6 +21,7 @@ envoy_cc_library(
         "//source/common/common:c_smart_ptr_lib",
         "//source/common/protobuf",
     ],
+    linkopts = ["-lluajit-5.1"],
```

Comment:
I would still argue that version numbers etc. don't belong in general `BUILD` files. Ideally we capture this kind of detail alongside the `repository_locations.bzl` or somewhere else confined to `bazel/` directory.

Reply:
I have even better idea. Since libluajit version may vary in different distributions, it might be reasonable to use `pkg-config` for outputing proper flags. Like i.e.:

```
$ pkg-config --libs luajit
-lluajit-5.1
```

I will try to write some macro or function for that in `bazel/` dir.

Reply:
Yeah, trying to hide this sounds nice, but using `pkg-config` is kind of non-hermetic, so that might be hard to fit into the Bazel model. I'd like to loop our resident Bazel expert @jmillikin-stripe into this discussion and also @irengrig from the Bazel development team who has been working on better integration with external build systems.

From a distribution perspective (i.e. SUSE) perspective, you probably don't even want to be using the version of LuaJIT that Envoy is pulling down from GitHub, you probably want to point at the existing installation in `/usr/lib`. This would involve setting up these external dependencies yourself in your WORKSPACE. I.e. before calling `envoy_dependencies()`, you already establish what `//external:luajit` and use he `skip_targets` arg.

Reply:
bazel allow toolchains to rely on system installed libararies, so if you want to use headers/libraries from /usr, you may just need to generate your own https://github.com/envoyproxy/envoy/blob/master/ci/prebuilt/BUILD file with correct path.

Reply:
@htuch Yes, being able LuaJIT and the other dependencies available in `/usr/lib` would be great.

@lizan OK, so I get that relying on system installed libraries is possible. Can you point me to some documentation or give some advice how should I generate my own BUILD file? Where exactly (in which directory) should I put this file?

Sorry for asking that basic questions, but using Google and looking at the documentation didn't make that clear for me and I feel lost.

Reply:
@lizan And another question. Is there something wrong with my attempt of changing `ci/prebuilt/BUILD` in this PR? `third_party/...` means `/usr/...` in practice, so I already teached this file to look for `.so` files in `/usr/lib*`.